# Standard components of a microbiome data science workflow

This session provides an overview of certain key elements of a typical
data science workflow in a taxonomic profiling study. These will be
mixed in various ways in real case studies.

You can use this to structure your own reproducible Rmarkdown report
by addressing the following questions by taking advantage of material
and examples in [OMA](https://microbiome.github.io/OMA/). Include
selected summaries and analyses of your choice, and focus on
delivering a clear and compact report.


## Alpha diversity

Estimate alpha diversity for each sample and draw a histogram. Tip:
estimateDiversity. Compare the results between two or more alpha
diversity indices (visually and/or statistically).


## Beta diversity

Visualize community variation with PCoA.  Investigate the influence of
the data transformations on statistical analysis: Visualize community
variation with PCoA with the following options: 1) Bray-Curtis
distances for compositional data; 2) Euclidean distances for
CLR-transformed data.

Community-level comparisons: Use PERMANOVA to investigate whether the
community composition differs between two groups of individuals
(e.g. males and females, or some other grouping of your choice). You
can also include covariates such as age and gender, and see how this
affects the results.


## Differential abundance and prevalent taxa

**Prevalence** What is the most prevalent genus in the data (tip:
getPrevalence and sort)

**Core microbiota** Pick up the core microbiota including taxa that
exceed 0.1% relative abundance in over 50% of the samples
(prevalence). How many core taxa there are? You can read more about
the core microbiota definition in [Salonen et
al. 2012](https://www.clinicalmicrobiologyandinfection.com/article/S1198-743X(14)60962-9/fulltext).
Visualize the core microbiota by following the available examples.

**Comparisons at the level of individual taxa** Use
[DESeq2](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8)
   to identify which genera are associated with gender differences
   ([examples on
   DESeq2](https://microbiome.github.io/tutorials/all.html)). For more
   explanation on the method, see [DESeq2
   R/Bioconductor](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html).

**Role of covariates** Note that including key covariates (diet,
medication, age..) may have remarkable influence on data
interpretation in the above analyses ([Falony et al. Science
2016](https://science.sciencemag.org/content/352/6285/560.abstract?ijkey=ADV4ZnF4mHYIg&keytype=ref&siteid=sci))

## Other material

Experiment with the other available tools in the [microbiome
tutorial](https://microbiome.github.io/tutorials/)

